### PR TITLE
Major renaming based on update to GEP 01.

### PR DESCRIPTION
**Pull request by ChristianZimpelmann**
### What problem do you want to solve?

Make the naming of functions, parameters, and files consistent over different modules and to (the clarifications of) [GEP 01](https://github.com/iza-institute-of-labor-economics/gettsim/pull/342).

### Todo

- [x] If it is a relevant contribution, put it in `CHANGES.rst`.
- [x] replace all Umlaute (e.g. `eink_st_abzuege` -> `eink_st_abzüge`)
- [x] higher order parts of a name come first (e.g. `proxy_eink_vorj_arbeitsl_geld` -> `arbeitsl_geld_proxy_eink_vorj`)
- [x] check consistency among modules 
- [x] month identifiers consistently used
- [x] alleinerziehend -> alleinerz

### open questions
- consistency between parameters <-> python files (see [discussion](https://github.com/iza-institute-of-labor-economics/gettsim/pull/342#discussion_r797447998))
- freibeträge: Hans-Martin: Please make a note for the big renaming thing that this is not great; will need to be more specific (add eink_st, make clear that it is just a shorthand for some, e.g., kinderfreib is not part of it).
- Rename `ges_rentenv.json` to `rente.json` (or `ges_rente.json`).  Reasoning: (Wir benutzen`ges_rentenv` als Abkürzung für Sozialbeiträge für die Rentenversicherung, aber `ges_rente_m`, `priv_rente_m` für die tatsächliche Rente)
- #346 (if nobody comes up with short German abbreviations, I am leaning towards English)
- #347 (not sure if we need to address in this PR)

### CODE REVIEW

File: gettsim/demographic_vars.py [hunk 1]
```diff
@@ -9,7 +9,9 @@
 from gettsim.typing import IntSeries
 
 
-def anz_minderj_hh(hh_id: IntSeries, alter: IntSeries, kind: BoolSeries) -> IntSeries:
+def anz_kinder_bis_18_hh(
```
Comment (hmgaudecker) [inline gettsim/demographic_vars.py:12]: Should this not be `anz_kinder_bis_17_hh` ? Apologies if I was responsible for that. If we write `bis_X`, it should be interpreted as `<=`.

**Comment by ChristianZimpelmann** [comment]
All proposed changes should be implemented now, as well as all other todos from the list above. The only remaining proposal seems to be this:

> - Rename `ges_rentenv.json` to `rente.json` (or `ges_rente.json`).  Reasoning: (Wir benutzen`ges_rentenv` als Abkürzung für Sozialbeiträge für die Rentenversicherung, aber `ges_rente_m`, `priv_rente_m` für die tatsächliche Rente)

I actually like `ges_rente.json`. What do you think?

**Comment by hmgaudecker** [comment]
> All proposed changes should be implemented now, as well as all other todos from the list above. The only remaining proposal seems to be this:
> 
> > Rename `ges_rentenv.json` to `rente.json` (or `ges_rente.json`).  Reasoning: (Wir benutzen`ges_rentenv` als Abkürzung für Sozialbeiträge für die Rentenversicherung, aber `ges_rente_m`, `priv_rente_m` für die tatsächliche Rente)
> 
> I actually like `ges_rente.json`. What do you think?

Sounds good, though these file renamings really don't matter much. We probably should do some major restructuring eventually, but they are not visible for users at all.

**Review by hmgaudecker — APPROVED** [review]
Great work, thanks!

**Comment by ChristianZimpelmann** [comment]
> Sounds good, though these file renamings really don't matter much. 

I made the change from `ges_rentenv.yaml` to `ges_rente.yaml` and will merge after the checks.

Thank you, too!

**Pull request merged**